stats/opentelemetry/client_metrics.go

```diff
@@ -163,10 +202,20 @@ func (h *clientStatsHandler) TagRPC(ctx context.Context, info *stats.RPCTagInfo)
 		}
 		ctx = istats.SetLabels(ctx, labels)
 	}
+	var ti *attemptTraceSpan
+	if !isTracingDisabled(h.options.TraceOptions) {
+		callSpan := trace.SpanFromContext(ctx)
+		if info.NameResolutionDelay {
+			callSpan.AddEvent("Delayed name resolution complete")
+		}
+		ctx = trace.ContextWithSpan(ctx, callSpan)
```

Comment:
can be one line `ctx, ti = h.traceTagRPC(trace.ContextWithSpan(ctx, callSpan), info)`